localstack-core/localstack/services/sns/provider.py

```diff
@@ -1116,7 +1118,25 @@ def _format_messages(sent_messages: List[Dict[str, str]], validated_keys: List[s
     return formatted_messages
 
 
-class SNSServicePlatformEndpointMessagesApiResource:
+class SNSInternalResource:
+    resource_type: str
+    """Base class with helper to properly track usage of internal endpoints"""
+
+    def count_usage(self):
+        usage.internalapi.record(f"{self.resource_type}")
+
+
+def count_usage(f):
+    @functools.wraps(f)
+    def _wrapper(self, *args, **kwargs):
+        self.count_usage()
+        return f(self, *args, **kwargs)
```

Comment:
perhaps we don't need `self`?

```suggestion
    def _wrapper(*args, **kwargs):
        self.count_usage()
        return f(*args, **kwargs)
```

Reply:
I think this will complain, we do need self passed, it failed before I added it 😄

Reply:
If I don't pass self, then I need to extract it from `args` right, otherwise `self.count_usage()` will raise an exception as it wouldn't be in scope?

Reply:
oh ,whoops, didn't see that there, of course your right!

Reply:
I'm not too fond of this either tbh, for a one line change I could just call `self.count_usage()` inside the method itself... it's the same as decorating it 😬